src/client/client.ts

```diff
@@ -468,6 +468,7 @@ export class Client implements Unsubscribable {
             if (!options.documentSelector && this.options.documentSelector) {
                 options.documentSelector = this.options.documentSelector
             }
+            options.providerName = this.name
```

Comment:
If the RegistrationParams already supplies a `providerName`, then it should not be clobbered here with `this.name`. Only if `options.providerName` is undefined should it be set to `this.name`. And when it is set to `this.name`, you should document that this is a fallback, like:

> No provider name was given, so use the client's name. (This might result in multiple providers from the same client having the same provider name. To avoid that, the server should supply distinct provider names.)